src/hotspot/cpu/aarch64/matcher_aarch64.hpp

```diff
@@ -163,6 +163,11 @@
   // Implements a variant of EncodeISOArrayNode that encode ASCII only
   static const bool supports_encode_ascii_array = true;
 
+  // Some architecture needs a helper to check for alltrue vector
+  static bool vectortest_need_second_argument(bool is_alltrue, bool is_predicate) {
```

Comment:
I think we can get the predicate feature in the backend (i.e.  `Matcher::has_predicated_vectors()`), and we do not need the argument here?

Reply:
On AVX3 the availability of predicated vectors does not ensure that every species of vector mask would be translated into predicated vector. To be more specific, although AVX512F returns true for `Matcher::has_predicated_vectors()`, byte and short vector masks required AVX512BW to be translated into predicated vectors and vector masks of < 512-bit species require AVX512VL.

Reply:
OK, I see, thanks!